### DISCUSSION TITLE
Linking multiple oath providers to same user

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [supabase/supabase#2127] [Questions] [answered]
Is it possible to link multiple oath providers to the same user? That way, the same user can sign in with either Twitter, Facebook, or Google?

It's possible to do this in Firebase: https://firebase.google.com/docs/auth/web/account-linking

**Comment by silentworks** [comment]
This feels like opening up multiple vectors of attack. I know most social log in websites normally allow you to sign in with only 1 provider. I don't think you can sign into multiple with using Supabase at the same time and if you could I would recommend against it for the same user.

**Reply by Nick-Mazuk** [reply]
That makes lots of sense. I didn't even think of that issue.

**Reply by Nick-Mazuk** [reply]
Interesting, it looks like Firebase implements this, though…

https://firebase.google.com/docs/auth/web/account-linking

**Reply by inian** [reply]
This looks like it is already implemented Nick https://github.com/supabase/gotrue/issues/81#issuecomment-871843225

Can you confirm and if there are any problems comment in that issue?

**Comment by jimmiejackson414-zz** [comment, marked as answer]
@Nick-Mazuk were you ever able to figure out a solution for this?

**Reply by Nick-Mazuk** [reply]
It works automatically: https://github.com/supabase/gotrue/issues/81#issuecomment-871843225
